src/operations/command.ts

```diff
@@ -73,12 +73,14 @@ export abstract class CommandOperation<
         : new MongoDBNamespace('admin', '$cmd');
     }
 
-    const propertyProvider = this.hasAspect(Aspect.NO_INHERIT_OPTIONS) ? undefined : parent;
+    const readPref = ReadPreference.fromOptions(options) ?? ReadPreference.primary;
```

Comment:
do you actually reuse `readPref` here? Should it just be inlined in the ternary?